Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -107,25 +105,53 @@ protected override string GetClassNameCore()
         /// </returns>
         protected override string GetNameCore()
         {
-            int? index = this.OwnerCarouselItem.ParentCarousel?.IndexFromContainer(this.OwnerCarouselItem);
+            string name = AutomationProperties.GetName(this.OwnerCarouselItem);
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
 
-            string name = base.GetNameCore();
+            name = this.OwnerCarouselItem.Name;
             if (!string.IsNullOrEmpty(name))
             {
-                return $"{name}";
+                return name;
             }
 
-            if (this.OwnerCarouselItem != null && !string.IsNullOrEmpty(this.OwnerCarouselItem.Name))
+            var textBlock = this.OwnerCarouselItem.FindDescendant<TextBlock>();
```

Comment:
We don't know there's going to be a textblock or anything, is this just a stab in the dark thinking that there may be some label added by a re-template of a developer?

Reply:
In a way, yes. This was to try and get some text to display mainly because the usage of the Carousel is likely to be a bound items source. This means that the container won't have a name property set. 

Alternatively, I could possibly make a change to the preparation of the container to set the name but this felt like it would be doing something that the developer might not be anticipating.